src/Data/Response.php

```diff
@@ -33,11 +35,15 @@ public function getRawData()
             'app_code'     => $this->getApplication()->getResponse()->getApplicationResponseCode(),
             'app_message'  => empty($appMessage) ? null : \json_encode($appMessage),
             'elapsed_time' => $this->getElapsedTime(),
-            'elapsed_time_ms' => (int) ($this->getElapsedTime() * 1000),
+            'elapsed_time_ms' => $responseElapsedTimeInMs,
             'profiler'     => \json_encode($profilerOutput),
             'app_version'  => $this->getAppVersionNumber(),
         ];
 
+        if($this->profiler->shouldLogProfilerOutput($responseElapsedTimeInMs)){
+            $rawData['profiler'] =  \json_encode($this->profiler->getProfilerOutput($httpCode));
```

Comment:
This problem is because the profiler output relies on $httpCode, not $responseElapsedTimeInMs.
This will not change the original behavior of the values 0, 1, or 2 (see application.ini)
You need to decide whether to log profiler output base don $httpCode and $responseElapsedTimeInMs.